### DISCUSSION TITLE
Version 0.7.1 / 0.10.1 Metrics not working [Windows]

### DISCUSSION THREAD

**Discussion by NoVa2294** [burningalchemist/sql_exporter#261] [Q&A] [answered]
Hi, I installed Version 0.7.1 as Windows Service because latest release is not working as a Service (Output 1067 error).
But the problem is that no metrics are showing when going to "localhost:1433/metrics" even with this version.
Port 1433 is discoverable with Test-NetConnection 192.xxx.x.xx -p 1433 and MSSQL ports are set to 1433 in SQL Server Configuration Manager.

Sql_exporter.yml : 
global:
  scrape_timeout_offset: 500ms
  min_interval: 0s
  max_connections: 3
  max_idle_connections: 3
  max_connection_lifetime: 5m
target:
  data_source_name: 'sqlserver://user:password@localhost:1433'
  collectors: [mssql_standard]
collector_files: 
  - "*.collector.yml"

Powershell Command : 
New-Service -name "SqlExporterSvc" `
-BinaryPathName "D:\Monitoring\sql_exporter-0.7.1.windows-amd64\sql_exporter.exe -config.file D:\Monitoring\sql_exporter-0.7.1.windows-amd64\sql_exporter.yml" `
-StartupType Automatic `
-DisplayName "SQL Exporter" `
-Description "SQL Exporter for Prometheus"

Idea behind the project is using Prometheus and Grafana to monitor with Windows_Exporter (Service) and SQL_Exporter (Service) multiple SQL Servers from I/O - CPU, RAM, Disks, etc. to MSSQL Db's and custom table values and queries.

SQL Server 14.0.2047.8 Version
Prometheus 2.44.0 Win
Grafana 9.5.2 Win

**Comment by burningalchemist** [comment]
Hi @NoVa2294, thanks for reporting the issue. I'll investigate the issue sql_exporter not working with Windows on `0.11.1`. Do you have the same configuration as for 0.7.1?

As for the errors, I suggest running the sql_exporter as a standalone binary (not as a service) to get more information about the errors. I need more context here to help. :)

**Comment by NoVa2294** [comment]
Hi @burningalchemist, yes I had the same configuration for 0.7.1 / 0.10.1 / 0.11.1 versions but just the 0.7.1 was a success as a running win service.

As for the stand alone binary running can you give more details on how to do it, I don't really know how to do that on windows.

Thanks! :)

**Comment by burningalchemist** [comment]
Hey @NoVa2294, got it. I'll take a look (need to unarchive my windows laptop) today.
To run the binary as a standalone process, just copy/paste the line you've shared to the Powershell (don't wrap it by the New-Service function):

```
D:\Monitoring\sql_exporter-0.7.1.windows-amd64\sql_exporter.exe -config.file D:\Monitoring\sql_exporter-0.7.1.windows-amd64\sql_exporter.yml
```

or just get to the directory via the Powershell and run `sql_exporter.exe` directly. If the configuration file is located at the same place, then it should start properly. Or exit with some error messages. Please paste them here.

This should start the server inside in your Powershell terminal. You can then try to call localhost:9399 and localhost:9399/metrics as well. All the errors will be send to the terminal too. By the way, I just noticed that you're probably using a wrong port to hit SQL Exporter endpoint. The server works on the port **9399** by default, not 1433.

**Comment by burningalchemist** [comment, marked as answer]
Hi @NoVa2294, I was able to successfully start and run version `0.11.1` on my old 2015 laptop with default settings. I guess you need to check your configuration and let's use double dash in front of `config.file`,like below:

```shell
New-Service -name "SqlExporterSvc" `
-BinaryPathName "%SQL_EXPORTER_PATH%\sql_exporter.exe --config.file %SQL_EXPORTER_PATH%\sql_exporter.yml" `
-StartupType Automatic `
-DisplayName "Prometheus SQL Exporter"
```

Both (single and double dash variants) were possible before, but it seems that Powershell doesn't like this format anymore and can't interpret the parameter correctly. I'll update the documentation. :)

Also, don't forget that you need to replace `%SQL_EXPORTER_PATH%` with a valid path to the binary and the config file.

After that the server starts listening on the port `9399`. I still suggest running the server in an interactive mode to debug potential errors before registering it as a Windows service.

Not a bug. I'll move it to Discussions. 👍
